src/Microsoft.Health.Fhir.Core/Extensions/LoggerExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿#nullable enable
+
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Core.Features.Parameters;
+
+namespace Microsoft.Health.Fhir.Core.Extensions
+{
+    public static class LoggerExtensions
+    {
+#pragma warning disable CA1068 // CancellationToken parameters must come last. Cancellation token being last breaks up the readability of the command by seperating the message and its args.
+        public static void LogVerbose(this ILogger logger, IParameterStore store, CancellationToken cancellationToken, string message, params object?[] args)
+#pragma warning restore CA1068 // CancellationToken parameters must come last
+        {
+            var logParameterTask = store.GetParameter("LogLevel", cancellationToken);
+
+            while (logParameterTask.Status != TaskStatus.RanToCompletion)
+            {
+                if (logParameterTask.Status == TaskStatus.Faulted)
+                {
+                    logger.LogWarning(logParameterTask.Exception, "Failed to retrieve LogLevel parameter. Logging at default level.");
+                    return;
+                }
+                else if (logParameterTask.Status == TaskStatus.Canceled)
+                {
+                    logger.LogWarning("Retrieving LogLevel parameter cancelled. Logging at default level.");
+                    return;
+                }
+
+                Thread.Sleep(5);
```

Comment:
I don't want this to be async because it prevents this logging type from being used in synchronous parts of the code. I feel logging calls should be as low impact as possible, if you have to do extra work for a logging call you aren't going to make the logging call (or you're going to use one of the logging types and this won't get used).
I don't like this, but there didn't seem to be any other way to get async calls to work in a synchronous method. All the built in ways to wait for a task throw exceptions if you actually try and use them :(

Reply:
I've also tested it with the cache, this sleep isn't hit once the value is in the local cache. So this will add 5ms to one call every 10 minutes.

Reply:
I see what you mean - Lazy does not work well with async code.